py/builtinimport.c

```diff
@@ -378,28 +378,22 @@ STATIC mp_obj_t process_import_at_level(qstr full_mod_name, qstr level_mod_name,
 
         // An import of a non-extensible built-in will always bypass the
         // filesystem. e.g. `import micropython` or `import pyb`.
-        module_obj = mp_module_get_builtin(level_mod_name);
+        module_obj = mp_module_get_builtin(level_mod_name, false);
         if (module_obj != MP_OBJ_NULL) {
             return module_obj;
         }
 
-        #if MICROPY_PY_SYS
-        // Never allow sys to be overridden from the filesystem. If weak links
-        // are disabled, then this also provides a default weak link so that
-        // `import sys` is treated like `import usys` (and therefore bypasses
-        // the filesystem).
-        if (level_mod_name == MP_QSTR_sys) {
-            return MP_OBJ_FROM_PTR(&mp_module_sys);
-        }
-        #endif
-
         // First module in the dotted-name; search for a directory or file
         // relative to all the locations in sys.path.
         stat = stat_top_level(level_mod_name, &path);
 
         // TODO: If stat failed, now try extensible built-in modules.
```

Comment:
The new code looks like it actually implements this, is this still TODO?

Reply:
Yep, I updated the comment without removing the TODO :)

Used this opportunity to re-work the comments here a bit further.